dask_match/tests/test_collection.py

```diff
@@ -139,12 +139,34 @@ def test_repr(ddf):
 
 
 def test_columns_traverse_filters(df, ddf):
-    result = optimize(ddf[ddf.x > 5].y)
+    result = optimize(ddf[ddf.x > 5].y, fuse=False)
     expected = ddf.y[ddf.x > 5]
 
     assert str(result) == str(expected)
 
 
+def test_optimize_fusion():
+    pdf = pd.DataFrame({"x": range(10), "y": range(10)})
+    ddf = from_pandas(pdf, npartitions=1)
+
+    ddf2 = (ddf["x"] + ddf["y"]) - 1
+    unfused = optimize(ddf2, fuse=False)
+    fused = optimize(ddf2, fuse=True)
+
+    # Should only get one task per partition
+    assert len(fused.dask) == ddf.npartitions
+    assert_eq(fused, unfused)
```

Comment:
@mrocklin - This test checks that we can handle the case you described here: https://github.com/mrocklin/dask-match/pull/11#issuecomment-1497413475

Reply:
I see!  🎉